app/assets/javascripts/discourse/controllers/preferences/interface.js.es6

```diff
@@ -50,6 +53,26 @@ export default Ember.Controller.extend(PreferencesTabController, {
     let key = this.get("themeKey");
     previewTheme(key);
   },
+  
+  homeChanged() {
+    let home = Discourse.SiteSettings.top_menu.split("|")[0].split(",")[0];
+    switch (Number(this.get('model.user_option.user_home'))) {
```

Comment:
Why not use a hash instead of a giant switch/case?

```
const USER_HOMES = { 1: "latest", 2: "categories", ..., 5: "top" };
homeChanged() {
  const home = USER_HOMES[this.get('model.user_option.user_home')] || Discourse.SiteSettings.top_menu.split("|")[0].split(",")[0];
  setDefaultHomepage(home);
}
`

Reply:
Good call.